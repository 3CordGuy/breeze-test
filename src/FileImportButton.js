import React, { useRef } from "react";
import { Button } from "semantic-ui-react";

const FileImportButton = ({
    onChange,
    text = "Choose File",
    accept = ".csv",
    value,
}) => {
    const inputFileRef = useRef(null);

    const handleClick = () => {
        inputFileRef.current.click();
    };

    return (
        <Button primary onClick={handleClick}>
            <input
                type="file"
                accept={accept}
                ref={inputFileRef}
                style={{ display: "none" }}
                value={value}
                onChange={(e) => onChange(e.target.files)}
            />
            {text}
        </Button>
    );
};

export default FileImportButton;
