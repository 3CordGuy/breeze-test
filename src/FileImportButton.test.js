import React from "react";
import { shallow } from "enzyme";
import FileImportButton from "./FileImportButton";

let wrapper;

describe("<FileImportButton />", () => {
    beforeAll(() => {
        wrapper = shallow(<FileImportButton />);

        wrapper.setProps({
            text: "It's dangerous to go alone... take this CSV file.",
        });
    });

    test("should match the snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
