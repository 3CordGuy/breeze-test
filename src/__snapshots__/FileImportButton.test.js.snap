// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileImportButton /> should match the snapshot 1`] = `
<Button
  as="button"
  onClick={[Function]}
  primary={true}
>
  <input
    accept=".csv"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  It's dangerous to go alone... take this CSV file.
</Button>
`;
